src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -263,7 +274,10 @@
   (fn [context]
     (let [desired-state (get-desired-state context)]
       (locking crl-write-serializer
-        (ca/set-certificate-status! settings subject desired-state)))))
+       (ca/set-certificate-status!
+        (merge-request-settings settings context)
```

Comment:
Do we want to do this inside the lock? Seems probably pretty trivial, but maybe the more we do outside it the better?

Reply:
@justinstoller was indifferent to this change, but he might have an opinion? It seems pretty trivial to me...

Reply:
Okay that's fine.